test/box-luatest/gh_11081_wal_queue_size_accounting_test.lua

```diff
@@ -0,0 +1,36 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function(g)
+    t.tarantool.skip_if_not_debug()
+    g.server = server:new()
+    g.server:start()
+end)
+
+g.after_all(function(g)
+    g.server:drop()
+end)
+
+-- Deliberately avoid dropping test space. It would hang.
+
+g.test_accounting_on_error = function(g)
+    g.server:exec(function()
+        local fiber = require('fiber')
+        local s = box.schema.create_space('test')
+        s:create_index('pk')
+        box.cfg{wal_queue_max_size = 100}
+        box.error.injection.set('ERRINJ_WAL_IO', true)
+        t.assert_error_msg_equals(
```

Comment:
Let's check the error name with `assert_error_covers` instead.

Reply:
Done.

```diff
         box.error.injection.set('ERRINJ_WAL_IO', true)
-        t.assert_error_msg_equals(
-            'Failed to write to disk',
-            s.insert, s, {1, string.rep('a', 1000)})
+        t.assert_error_covers({
+            type = "ClientError",
+            code = 40,
+            message = 'Failed to write to disk',
+        }, s.insert, s, {1, string.rep('a', 1000)})
         box.error.injection.set('ERRINJ_WAL_IO', false)
```

Reply:
Please change to

```lua
code = box.error.WAL_IO,
```

or

```lua
name = 'WAL_IO',
```